### PR TITLE
Introduce SystemTableProvider

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 1]
```diff
@@ -467,7 +470,7 @@ public ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSessi
         if (SystemTableHandler.PROPERTIES.matches(tableName)) {
             return getPropertiesSystemTable(session, tableName, SystemTableHandler.PROPERTIES.getSourceTableName(tableName));
         }
-        return Optional.empty();
+        return additionalSystemTableProvider.flatMap(provider -> provider.getSystemTable(session, tableName));
```
Comment (losipiuk) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:473]: Instead can you extract system tables we are exposing currently as a `SystemTablesProvider`. And then bind Set<SystemTablesProvider> in connector.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 2]
```diff
@@ -344,6 +345,7 @@ public HiveMetadata(
             String trinoVersion,
             HiveStatisticsProvider hiveStatisticsProvider,
             HiveRedirectionsProvider hiveRedirectionsProvider,
+            Optional<SystemTableProvider> additionalSystemTableProvider,
```
Comment (raunaqmorarka) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:348]: Can we make it `Set<SystemTableProvider>` instead and return PARTITIONS/PROPERTIES also through a provider ?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 1 above)
Comment (sopel39) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:473]: Current system tables make use of `SemiTransactionalHiveMetastore` which is created by `io.trino.plugin.hive.HiveMetadataFactory#create()`. I didn't really want to complicate stuff for the same of extension

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 2 above)
Comment (sopel39) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:348]: see https://github.com/trinodb/trino/pull/8476#discussion_r664485191

**Review by losipiuk — APPROVED** [review]

**Review by raunaqmorarka — APPROVED** [review]

**Pull request merged**